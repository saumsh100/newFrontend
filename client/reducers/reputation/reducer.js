import { fromJS } from 'immutable';

import {
  FETCH_REPUTATION,
  FETCH_REPUTATION_SUCCESS
} from './constants';

const initialState = fromJS({
  status: 'loading', // set only to 'loading', 'success' or 'error'
  lastFetched: null,
  data: {
    sourcesNotFound: null,
    sourcesFound: null,
    listingScore: null,
    listingPointScore: {
      industryAverage: null,
      industryLeadersAverage: null,
      pointScore: null
    },
    citationsFound: null,
    sourcesFoundWithErrors: null,
    accuracyScore: null
  }
})

function reputationReducer(state = initialState, action) {
  switch (action.type) {
    case FETCH_REPUTATION:
      return state.set('status', 'loading')
    case FETCH_REPUTATION_SUCCESS:
      return state
        .set('status', 'success')
        .set('lastFetched', (new Date()).toLocaleString())
        .set('data', action.data)
    default:
      return state
  }
}

export default reputationReducer;