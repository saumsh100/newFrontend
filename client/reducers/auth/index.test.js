import { fromJS } from 'immutable';
import reducer, {
  initialState,
  loginSuccess,
  authLogout,
  adapterPermissionsSet,
  isAdapterPermissionEnabledSelector,
} from './index';

describe('auth reducer', () => {
  const loginSuccessAction = loginSuccess({
    userId: '6603fe66-aa03-4ffe-8de6-ce07a4ae9c5c',
    role: 'SUPERADMIN',
    enterpriseId: 'f1405909-55cf-422a-9ec0-9f80de72b0bf',
    accountId: 'c05ade0e-eb8a-44ed-bf3c-a83bfa4bc5a3',
    enterprise: {
      id: 'f1405909-55cf-422a-9ec0-9f80de72b0bf',
      name: 'Gallery Dental Inc.',
      plan: 'ENTERPRISE',
      createdAt: '2018-04-11T17:19:13.773Z',
      updatedAt: '2018-04-11T17:19:13.773Z',
      deletedAt: null,
    },
    user: {
      id: '6603fe66-aa03-4ffe-8de6-ce07a4ae9c5c',
      firstName: 'Justin',
      lastName: 'Sharp',
      username: 'justin@carecru.com',
    },
    sessionId: 'd15fbddc-f306-49e1-b736-78cb6a6166d4',
  });

  const logoutAction = authLogout(undefined);

  const adapterPermissionsSetAction = adapterPermissionsSet({
    'can-add-patient': true,
    'can-update-appointment': false,
    'can-set-ismissing': true,
  });

  test('action creator works', () => {
    expect(loginSuccessAction).toMatchSnapshot();
    expect(logoutAction).toMatchSnapshot();
    expect(adapterPermissionsSetAction).toMatchSnapshot();
  });

  test('selectors works', () => {
    const adapterPermissions = fromJS(adapterPermissionsSetAction.payload);
    expect(isAdapterPermissionEnabledSelector(adapterPermissions, 'can-add-patient')).toEqual(true);
    expect(
      isAdapterPermissionEnabledSelector(adapterPermissions, 'can-update-appointment'),
    ).toEqual(false);
    expect(isAdapterPermissionEnabledSelector(adapterPermissions, 'can-set-ismissing')).toEqual(
      true,
    );
  });
  test('reducer works', () => {
    expect(reducer(initialState, {})).toEqual(initialState);
    expect(reducer(initialState, loginSuccessAction)).toMatchSnapshot();
    expect(reducer(initialState, logoutAction)).toMatchSnapshot();
    expect(reducer(initialState, adapterPermissionsSetAction)).toMatchSnapshot();
  });
});
