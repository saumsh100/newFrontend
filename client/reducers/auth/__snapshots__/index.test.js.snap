// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducer action creator works 1`] = `
Object {
  "payload": Object {
    "accountId": "c05ade0e-eb8a-44ed-bf3c-a83bfa4bc5a3",
    "enterprise": Object {
      "createdAt": "2018-04-11T17:19:13.773Z",
      "deletedAt": null,
      "id": "f1405909-55cf-422a-9ec0-9f80de72b0bf",
      "name": "Gallery Dental Inc.",
      "plan": "ENTERPRISE",
      "updatedAt": "2018-04-11T17:19:13.773Z",
    },
    "enterpriseId": "f1405909-55cf-422a-9ec0-9f80de72b0bf",
    "role": "SUPERADMIN",
    "sessionId": "d15fbddc-f306-49e1-b736-78cb6a6166d4",
    "user": Object {
      "firstName": "Justin",
      "id": "6603fe66-aa03-4ffe-8de6-ce07a4ae9c5c",
      "lastName": "Sharp",
      "username": "justin@carecru.com",
    },
    "userId": "6603fe66-aa03-4ffe-8de6-ce07a4ae9c5c",
  },
  "type": "@auth/LOGIN_SUCCESS",
}
`;

exports[`auth reducer action creator works 2`] = `
Object {
  "payload": undefined,
  "type": "@auth/LOGOUT",
}
`;

exports[`auth reducer action creator works 3`] = `
Object {
  "payload": Object {
    "can-add-patient": true,
    "can-set-ismissing": true,
    "can-update-appointment": false,
  },
  "type": "@auth/ADAPTER_PERMISSIONS_SET",
}
`;

exports[`auth reducer reducer works 1`] = `
Immutable.Map {
  "enterprise": Immutable.Map {
    "id": "f1405909-55cf-422a-9ec0-9f80de72b0bf",
    "name": "Gallery Dental Inc.",
    "plan": "ENTERPRISE",
    "createdAt": "2018-04-11T17:19:13.773Z",
    "updatedAt": "2018-04-11T17:19:13.773Z",
    "deletedAt": null,
  },
  "isAuthenticated": true,
  "userId": "6603fe66-aa03-4ffe-8de6-ce07a4ae9c5c",
  "accountId": "c05ade0e-eb8a-44ed-bf3c-a83bfa4bc5a3",
  "user": Immutable.Map {
    "id": "6603fe66-aa03-4ffe-8de6-ce07a4ae9c5c",
    "firstName": "Justin",
    "lastName": "Sharp",
    "username": "justin@carecru.com",
  },
  "sessionId": "d15fbddc-f306-49e1-b736-78cb6a6166d4",
  "adapterPermissions": Immutable.Map {},
  "role": "SUPERADMIN",
  "enterpriseId": "f1405909-55cf-422a-9ec0-9f80de72b0bf",
  "forgotPassword": false,
}
`;

exports[`auth reducer reducer works 2`] = `
Immutable.Map {
  "isAuthenticated": false,
  "forgotPassword": false,
  "role": null,
  "accountId": null,
  "enterprise": null,
  "user": null,
  "sessionId": null,
  "adapterPermissions": Immutable.Map {},
}
`;

exports[`auth reducer reducer works 3`] = `
Immutable.Map {
  "isAuthenticated": false,
  "forgotPassword": false,
  "role": null,
  "accountId": null,
  "enterprise": null,
  "user": null,
  "sessionId": null,
  "adapterPermissions": Immutable.Map {
    "can-add-patient": true,
    "can-update-appointment": false,
    "can-set-ismissing": true,
  },
}
`;
